### DISCUSSION TITLE
Using @nx/js:tsc Batch Mode for Type Checking

### DISCUSSION THREAD

**Discussion by bacarybruno** [nrwl/nx#21989] [General]
Hello everyone 👋
I'm working on a large monorepo and was exploring some ways to speed up the build time.

While investigating potential solutions, I came across @nx/js:tsc batch mode, which seemed promising. I'm particularly interested in running multiple tasks in a single process & leveraging project references. Also incremental mode but I can already do that by passing the right flags.

However, it appears to focus on compilation rather than type checking. I'm curious if there's a method to use this mode for type checking, perhaps by running it in noEmit mode and conducting type checks across the entire workspace, rather than relying on a  `main` field. Any insights on this?

**Comment by eliasm307** [comment]
Just came across this as I was investigating/thinking the same, this would be useful if possible
